scripts/zephyr_module.py

```diff
@@ -119,12 +119,14 @@ def main():
                              'list`')
     parser.add_argument('-x', '--extra-modules', nargs='+',
                         help='List of extra modules to parse')
+    parser.add_argument('-w', '--west-path', help='Path to west executable')
     args = parser.parse_args()
 
     if args.modules is None:
-        p = subprocess.Popen(['west', 'list', '--format={posixpath}'],
+        p = subprocess.Popen([args.west_path, 'list', '--format={posixpath}'],
```

Comment:
This breaks when (ZEPHYR_MODULES AND NOT WEST), doesn't it?

Reply:
Shouldn't as `if args.modules is None` checks for `ZEPHYR_MODULES` but I could be wrong

Reply:
My mistake.

Reply:
I however do think that one could  consider changing the check to
```python
if args.west_path:
```
as it's a bit of readability/logical issue as we keep checking if not modules use west. While maybe one should check for if *west* use `west` if *modules* use `modules` else throw an error.

Reply:
This breaks if running the script and not supplying `--west-path` with following error.
```
./scripts/zephyr_module.py --cmake-out c.out --kconfig-out k.out
Traceback (most recent call last):
  File "./scripts/zephyr_module.py", line 167, in <module>
    main()
  File "./scripts/zephyr_module.py", line 128, in main
    stderr=subprocess.PIPE,
  File "/usr/lib/python3.6/subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.6/subprocess.py", line 1258, in _execute_child
    executable = os.fsencode(executable)
  File "/usr/lib/python3.6/os.py", line 800, in fsencode
    filename = fspath(filename)  # Does type-checking of `filename`.
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

You are making it mandatory to give `--west-path`.
While I think `--west-path` is a good option for CMake, I don't see why it should restrict people from running the script by hand and use the `west` tool in path.

So please keep a fallback to use `west` from path if not given by `--west-path`

Reply:
set `default='west'`